### PR TITLE
feat: Metadata panel v1

### PR DESCRIPTION
**Pull request by bprusinowski**
TODO:
- [ ] add expanding of long dimension descriptions + autocomplete
- [ ] add a way of opening the panel by clicking on values in chart
- [ ] add shrinking of chart after the panel is opened in published / embed mode
- [ ] add tests
- [ ] do not show `Data` tab when no metadata is attached to dimensions
- [ ] add animations
- [ ] dimension values?

### CODE REVIEW

File: app/components/metadata-panel.tsx [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  children: React.ReactNode;
+}) => {
+  const [open, setOpen] = React.useState(false);
+
+  const ctx = React.useMemo(() => {
+    return {
+      open,
+      toggle: () => setOpen(!open),
+    };
+  }, [open, setOpen]);
```
Comment (ptbrowne) [inline app/components/metadata-panel.tsx]: useDisclosure is helpful for those usecases.

File: app/icons/index.tsx [hunk 2]
```diff
@@ -44,3 +53,23 @@ export const getChartIcon = (chartType: ChartType): IconName => {
...
+      return "chartPie";
+    case "NumericalMeasure":
+      return "numerical";
+    case "OrdinalDimension":
+      return "sort";
+    case "OrdinalMeasure":
+      return "sort";
+    case "TemporalDimension":
+      return "pointInTime";
+  }
```
Comment (ptbrowne) [inline app/icons/index.tsx:74]: could we have a exhaustive check here ?

I think `return dimensionType as never` should work.

**Review by ptbrowne — APPROVED** [review]
👏  I think we could merge if we used a feature flag to activate / deactivate the button to open the metadata panel ?

File: app/components/metadata-panel.tsx [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    <ContextProvider>
+      <PanelInner />
+    </ContextProvider>
+  );
+};
+
+const ToggleButton = ({ onClick }: { onClick: () => void }) => {
+  return (
+    <Button component="a" variant="text" size="small" onClick={onClick}>
+      <Typography variant="body2">Metadata</Typography>
```
[the remark this answers is not in the document]
Comment (bprusinowski) [reply app/components/metadata-panel.tsx:225]: I think we could maybe make it bold when the panel is open?

cc @AnninaWalker – it's about the button below :)

Screenshot 2022-11-28 at 09 36 09

**Review by ptbrowne — APPROVED** [review]

**Pull request merged**